pkg/pop/mix.cue

```diff
@@ -0,0 +1,49 @@
+import "encoding/json"
+
+// TODO: include full schema?
+// TODO: figure out why it stops working when uncommented (cue eval works tho)
+// prometheusAlerts: v0: {...}
+// prometheusRules: v0: {...}
+// grafanaDashboards: v0: {...}
+
+// TODO: handle different versions, deal with migrations, etc, etc
+mixin: {
+    // TODO: change to ruleConvert() once syntax sugar is available
+    _alerts: #ruleConvert&{_, #rules: prometheusAlerts.v0}
+    _rules: #ruleConvert&{_, #rules: prometheusRules.v0}
+    _dashboards: #dashboardConvert&{#dashboards: grafanaDashboards.v0}
+
+    {
+        prometheusAlerts: _alerts
+        prometheusRules: _rules
+        grafanaDashboards: _dashboards
+    }
+}
+
+#ruleConvert: {
+    #rules: [string]: {group: string, alert: {...}, ...}
+
+    [for n, r in (#flattenAlerts & {arg: #rules}).out {
+        name: n, rules: r
+    }]
+}
+
+#dashboardConvert: {
+    #dashboards: [string]: _
+
+    for k, v in #dashboards {
+        "\(k).json": json.Indent(json.Marshal(v), "", "  ")
+    }
+}
+
+#flattenAlerts: {
```

Comment:
why not import the one that's already written and exported?

Reply:
bad argument: I did not manage to get imports working

good argument: whatever the binary needs should be self contained, we should not rely on the users cue.mod to have what we need. that also makes it predictable

Reply:
Generally agreed, but i think that for this case, it's fine to embed everything contained in the same CUE module and thereby treat those imports as safe. By CUE's own rules, those don't trigger package management.

In fact, i'd even argue that not including the entire tree that shares an import namespace/`cue.mod` actually violates the principle of least surprise inasmuch as it disallows same-module imports.